### PR TITLE
[Form] switch the context when validating nested forms

### LINKED ISSUE #37072: [SymfonyForm] Updating to 4.4.9 crash with Unable to get a property on a non-object.
**Symfony version(s) affected**: 4.4.9

**Description**  

I'm using a childAdmin with Sonata 
https://symfony.com/doc/master/bundles/SonataAdminBundle/reference/child_admin.html

When I'm on a playList/idPlaylist/video/create url and try to submit the form to create a video and add it to the playlist, I get `Unable to get a property on a non-object.`.

Everything worked fined with symfony-form@4.4.8, but when I update to 4.4.9 I get an error.

**Additional context**  
image

### PR DESCRIPTION
**Pull request by xabbuh**
| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       | Fix #37072
| License       | MIT
| Doc PR        |

### CODE REVIEW

**Comment by xabbuh** [comment]
tests added

Status: Needs Review

**Review by VincentLanglet — APPROVED** [review]

**Comment by fabpot** [comment]
Tests look broken by this change.

File: src/Symfony/Component/Form/composer.json [hunk 1]
```diff
@@ -26,9 +26,10 @@
     },
     "require-dev": {
         "doctrine/collections": "~1.0",
-        "symfony/validator": "^3.2.5|~4.0",
+        "symfony/validator": "^3.4.3|^4.0.3",
```
Comment (xabbuh) [inline src/Symfony/Component/Form/composer.json:29]: to get #25529 which prevents the `deps=low` build to fail

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @xabbuh.

**Pull request merged**